### ISSUE TITLE
Finish refactoring Creature Constructors

### ISSUE THREAD

**Issue by yueranyuan** [CompletelyFairGames/dwarfcorp#13]
It looks like the Creature constructor and EntityFactory code are in the process of being refactored to loading CreatureDef objects from JSON. This seems nice and can eliminate quite a bit of redundant code.

Doesn't seem like a high priority issue but since the code is already written, we should probably bring all the creatures over to using CreatureDef eventually.

**Comment by Solsund** [comment]
I have to agree with this.  There are comments in the code talking about how some things were done in a more complex manner than otherwise needed to make it easier to mod things down the line.  Having things hard coded just makes this process worse.

**Comment by mklingen** [comment]
I actually forgot about the existence of CreatureDef. I would like to do this eventually but it is not as easy as it looks. For one, I started using the C# hardcoded creature definitions to extend behavior. For example, the Necromancer has special AI that allows it to spawn Skeletons. Other creatures have similar ad-hoc extensions. If we want creatures to be loaded from files, we will have to push all of those special extensions up to Creature and represent them in the data.

That said, there is a *lot* in the game as it stands which can and should be converted into JSON data files. These would be:

- The classes/levels (worker, soldier, etc.) which are defined in numerous annoying C# classes right now
- The particle effects (currently initialized in *wait for it* PlayState.cs)
- The definition of voxels (in VoxelLibrary.cs)
- Room definitions (RoomLibrary.cs)
- Basically anything that's in a *Library.cs. I deliberately named the classes this way because I anticipated they would eventually be replaced by data files.
- The big cahuna would be EntityFactory.cs, whose only reason for existing is to map strings to functions which create entities. I'm not sure how to turn that into a data file, but that would clean up the code base tremendously.

Actually getting that done would not be too difficult. It would just involve saving the existing structures to JSON, and then deleting all the hardcoded crap and replacing it with a file load.
